src/core/src/pass/stateful_to_stateless.cpp

```diff
@@ -0,0 +1,172 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "openvino/pass/stateful_to_stateless.hpp"
+
+#include <regex>
+#include <string>
+
+#include "openvino/cc/pass/itt.hpp"
+#include "openvino/op/assign.hpp"
+#include "openvino/op/gather.hpp"
+#include "openvino/op/read_value.hpp"
+#include "openvino/pass/manager.hpp"
+#include "transformations/utils/utils.hpp"
+
+using namespace ov::op;
+
+namespace {
+
+std::shared_ptr<ov::Node> set_name(std::shared_ptr<ov::Node> node, const std::string& name) {
+    // Set name for both node and output tensor (should be only one tensor, and any other names will be overriden by a
+    // given single name)
+    node->set_friendly_name(name);
+    OPENVINO_ASSERT(node->get_output_size() == 1);
+    node->get_output_tensor(0).set_names({name});
+    return node;
+}
+
+// Templated method that has the same effect as not templated `set_name` but saves Op type for convenient calls chaining
+template <typename T>
+inline std::shared_ptr<T> set_name(std::shared_ptr<T> node, const std::string& name) {
+    set_name(std::dynamic_pointer_cast<ov::Node>(node), name);
+    return node;
+}
+
+std::shared_ptr<v0::Parameter> get_parameter_by_tensor_name(const std::shared_ptr<ov::Model>& model,
+                                                            const std::string& name) {
+    for (const auto& param : model->get_parameters()) {
+        if (param->get_output_tensor(0).get_names().count(name))
+            return param;
+    }
+    return nullptr;  // nullptr and return type are only difference from ov::Model::input(name)
+}
+
+struct Variable {
+    struct Context {
+        // to hold compiled once regex for all Variable instances
+        const std::regex naming_convention =
```

Comment:
Is it possible make it static member of Variable structure and do not pass it to Variable ctor

Reply:
No, because it won't have a bounded lifetime. I want to keep it alive only during the transformation run time because it is needed only once in most cases.